src/components/admin/form-components/SearchBar.jsx

```diff
@@ -49,45 +84,40 @@ export class SearchBar extends React.Component {
             style={this.props.fullWidth ? { width: '100%' } : { width: 200 }}
           >
             {this.props.enableAdd &&
-              this.props.suggestions.length === 0 &&
-              this.state.value && (
+              this.state.suggestions.length === 0 &&
+              this.state.value.trim() && (
                 <div
                   className={`${searchBarResultClass} search-bar-${
-                    this.props.category
+                    this.props.mode
                   }`}
-                  key={slug}
+                  key={this.props.slugify(this.state.value.trim())}
                 >
-                  {/* eslint-disable react/jsx-no-bind */}
-                  <MenuItem
-                    leftIcon={<ContentAdd />}
-                    primaryText={this.state.value}
-                    onClick={this.handleClick.bind(this, {
-                      title: this.state.value,
-                      slug,
-                      id: null,
-                    })}
+                  <SearchResult
+                    id={null}
+                    title={this.state.value.trim()}
+                    slug={this.props.slugify(this.state.value.trim())}
+                    handleClick={this.handleClick}
                     disabled={this.props.disabled}
+                    enableAdd={this.props.enableAdd}
```

Comment:
Quickly looking through this code I got the fleeting impression, mostly because of variable names and the fact that you pass this prop down, that all the suggestions were gonna be addable, I think an easy thing to do would be to call it `isNew` or something like that and hardcode in the true here (as that's also the real effect of this line right, it should always be true as you check the value of `enableAdd` in the conditional)

Reply:
@emilgoldsmith can you elaborate on what you mean? The enableAdd feature is a feature that allows you to add (primarily) tags that don't already exist. If enableAdd is not set, then no such add option appears.

Reply:
Yeah no I agree that the javascript and the actual logic is correct, I just didn't find it way too readable. It's just for the `SearchResult` if we could hardcode in the true (as if it shows up it should always be true), and I personally found it a bit harder to read when that prop was also called enableAdd like it is in the parent component, so I was thinking of renaming it `isNew` or something like that. Could possibly just be me, this one doesn't block me from merging it, so you can take my input and do what you want with it.

If @Helw150 has nothing to do while recovering from his marathon also feel free to give a third opinion (I know our discussions on naming are always legendary :P)